### PR TITLE
NT-1735: Target API 30 🚀 

### PR DESCRIPTION
**Pull request by leighdouglas**
# 📲 What

Upgrading to Android API 30 requires a few updates for our codebase to be compatible. I have added a queries element to the Android manifest file for the new package visibility requirements in API 30. Also updated tests to reflect new formatting changes.

- Added a new queries element to android manifest.
- Updated the compileSdkVersion and targetSdkVersion to 30
- Upgraded robolectric dependency to 4.4
- Retrieving a project object from an intent now checks for nullability
- Set KSCurrency with a mock current config to the mock environment object used for testing view models to pass tests failing on the external release variant
- Disabled pausing and unpausing the shadow looper in the `ObserveForUITransformerTest` suite (more on that below)
- Amended the jacoco config struct to exclude coverage for internal jdk classes -> this is necessary since we are using JDK 1.8 (more on that below)
- Updated the codecov config struct to be informational and not pass/fail PRs
- Updated failing tests with new formatting to Date and Number strings:
```
Dec 17, 2015 6:35 PM -> Dec 17, 2015, 6:35 PM
17 déc. 2015 18:35:05 -> 17 déc. 2015 à 18:35:05
100 $ -> now has a non-breaking space
```

# 🛠 How

More context on `ObserveForUITransformerTest` -> Robolectric 4.4 was a massive update that included changes to their shadow looper. These changes are causing mysterious test failures during the `tearDown()` that happens after the test is run, specifically it is throwing an `java.lang.UnsupportedOperationException: main looper cannot be unpaused` exception. After exhausting a few solutions from their documentation (they have a whole article about it here-> http://robolectric.org/blog/2019/06/04/paused-looper/), and since all of this code is about 5 years old, @jgsamudio and I decided to remove these lines since the tests are passing without them. Perhaps robolectric is not performing these operations for us under the hood. 

For the jacoco config to exclude generating coverage for jvm internal classes, this is a known issue and related to our target jvm being 1.8. See -> https://github.com/gradle/gradle/issues/5184

# 👀 See

No UI changes

# 📋 QA

Since these are all internal changes and not user facing, I would test this by opening the app and making sure nothing is crashing. 

# Story 📖

[[SPIKE] Fix Test Environment for Android 11 Migration](https://kickstarter.atlassian.net/browse/NT-1735)
[[Spike]Android 11: Target API 30](https://kickstarter.atlassian.net/browse/NT-1626)

### CODE REVIEW

File: app/src/test/java/com/kickstarter/KSRobolectrictTestCase.kt [hunk 1]
```diff
@@ -49,7 +51,13 @@ abstract class KSRobolectricTestCase : TestCase() {
             .applicationModule(TestApplicationModule(application()))
             .build()
 
+        val config = ConfigFactory.config().toBuilder()
```
Comment (Arkariang) [inline app/src/test/java/com/kickstarter/KSRobolectrictTestCase.kt:54]: nice!

File: app/src/test/java/com/kickstarter/libs/rx/transformers/ObserveForUITransformerTest.java [hunk 2]
```diff
@@ -23,13 +23,13 @@
   @Before
   public void setUp() {
     RxAndroidPlugins.getInstance().reset();
-    ShadowLooper.pauseMainLooper();
+    //ShadowLooper.pauseLooper(Looper.myLooper());
```
Comment (Arkariang) [inline app/src/test/java/com/kickstarter/libs/rx/transformers/ObserveForUITransformerTest.java:26]: delete this two lines please

**Review by Arkariang — APPROVED** [review]
Great work here @leighdouglas!

File: app/src/test/java/com/kickstarter/viewmodels/BackingFragmentViewModelTest.kt [hunk 3]
```diff
@@ -94,6 +96,7 @@ class BackingFragmentViewModelTest : KSRobolectricTestCase() {
         this.vm.outputs.projectDataAndAddOns().subscribe(this.listAddOns)
         this.vm.outputs.bonusSupport().map { it.toString() }.subscribe(this.bonusAmount)
         this.vm.outputs.deliveryDisclaimerSectionIsGone().subscribe(this.disclaimerSectionIsGone)
+        Shadows.shadowOf(Looper.getMainLooper()).idle()
```
Comment (Arkariang) [inline app/src/test/java/com/kickstarter/viewmodels/BackingFragmentViewModelTest.kt:99]: I think these `Shadows.shadowOf(Looper.getMainLooper()).idle()` are no longer necessary as you found the root cause if blocking problem

File: app/src/test/java/com/kickstarter/libs/rx/transformers/ObserveForUITransformerTest.java (hunk 2 above)
Comment (leighdouglas) [reply app/src/test/java/com/kickstarter/libs/rx/transformers/ObserveForUITransformerTest.java]: Done 👍🏽

File: app/src/test/java/com/kickstarter/viewmodels/BackingFragmentViewModelTest.kt (hunk 3 above)
Comment (leighdouglas) [reply app/src/test/java/com/kickstarter/viewmodels/BackingFragmentViewModelTest.kt]: Done, and everything passed CI! 🎉 🚀

**Pull request merged**